game-app/game-headed/src/main/java/games/strategy/engine/framework/startup/ui/panels/main/HeadedServerSetupModel.java

```diff
@@ -100,7 +101,13 @@ public void onServerMessengerCreated(
    */
   public void showClient() {
     Preconditions.checkState(!SwingUtilities.isEventDispatchThread());
-    final ClientModel model = new ClientModel(gameSelectorModel, this, new HeadedLaunchAction(ui));
+    final ClientModel model =
+        new ClientModel(
+            gameSelectorModel,
+            this,
+            new HeadedLaunchAction(ui),
+            HeadedGameRunner::showMainFrame,
```

Comment:
Any reason these are passed separately instead just being methods on the launch action?

Reply:
Mainly because there's no headless counterpart from what I can tell. To be fair many methods are rather close to no-op so that's a rather weak argument. To some degree I see this solution as temporary glue until `ClientModel` can be moved to `game-headed` which would allow us to just remove this argument again